idaes/models_extra/power_generation/properties/natural_gas_PR.py

```diff
@@ -75,18 +87,27 @@ class EosType(enum.Enum):
         "type": VaporPhase,
         "equation_of_state": Cubic,
         "equation_of_state_options": {"type": CubicType.PR},
+        "visc_d_phase": ViscosityWilke,
+        "therm_cond_phase": ThermalConductivityWMS,
     },
     "Liq": {
         "type": LiquidPhase,
         "equation_of_state": Cubic,
         "equation_of_state_options": {"type": CubicType.PR},
+        "visc_d_phase": None,
```

Comment:
This made me think of a possible option for the issue of skipping properties in certain phases - maybe we could have a `NoMethod` option instead of `None`. We would still insist that users provide a method for all properties in all phases, but for cases like this it would let the user flag a given phase to be skipped - the benefit being that it is explicit in the config dictionary.

Reply:
That is probably less important, as they are semi-private at least, although being explicit is always good.

Reply:
I think those are best left private - they have questionable physical meaning anyway, even when they are all defined.

Reply:
Wait, this doesn't work the way I thought it would. I was hoping to be able to do something like:

```
    class therm_cond_phase(object):
        """No method for thermal conductivity"""

        @staticmethod
        def build_parameters(pobj):
            """No parameters needed"""
            pass

        @staticmethod
        def return_expression(b, p):
            """Raise error when therm_cond_phase is called for in phase p"""
            raise PropertyNotSupportedError(
                f"Property package does not support thermal conductivity in phase {p}"
            )
```

Unfortunately, that error is just going to get raised during `Expression` creation, not when the user tries to access it. If we're just hiding the `Expression.Skip` inside a new class, is that so much better than what we had before?

Reply:
What I had in mind was this:

```
 class therm_cond_phase(object):
        """No method for thermal conductivity"""

        @staticmethod
        def build_parameters(pobj):
            """No parameters needed"""
            pass

        @staticmethod
        def return_expression(b, p):
            """Raise error when therm_cond_phase is called for in phase p"""
            _log.debug(f"Skipping construction of thermal conductivty for phase {p}")
            return Expression.Skip
```

Yes, it does do exactly the same thing, and just buries the `Expression.Skip` deeper. However, it does add two things:

1. It adds a logger message to help users who run into issue because of this (if they care to look).
2. More importantly, it makes the decision to skip that phase explicit in the configuration dictionary; i.e., the developer has to clearly state that they are skipping this phase. this is better than a system where it can be skipped implicitly by just not declaring anything.

Basically, the point is to force the developer/user to acknowledge the choice, and to provide hints to later users if something goes wrong. Unfortunately, I do not think we can (easily) deal with the actually error that will occur if/when someone tries to access a skipped property.

Reply:
(2) was true of the old method as well. (1) is potentially useful, but I think any user who pays attention to their log is probably going to be able to figure out the Pyomo error anyway.

Reply:
The issue with using `None` is that is can be done implicitly is you are not careful (e.g. it is the default value for `ConfigValues`, and is generally accepted as a value in any `ConfigValue` as well). It is better to use something explicit that cannot be confused for not providing any value.

As for 1, you over estimate the ability of our general users. However, the main purpose here is not to provide a hint; it is so we have something to point to when someone complains (i.e. the logger told you what was wrong, but you ignored it).

Reply:
Made a [feature request for Pyomo](https://github.com/Pyomo/pyomo/issues/2770#issue-1629760340) to maybe let us attach additional information to the `KeyError` raised, or maybe substitute our own exception. In the meantime, we'll log it.